docs/lsm/embedded_entities/embedded_entities.rst

```diff
@@ -208,6 +230,46 @@ migration from cluster A to cluster B:
 
 For these more involved update scenarios we recommend updating the lifecycle specifically for this update.
 
+.. _using_custom_lifecycle:
+
+
+Tracking embedded entities when using a custom lifecycle
+********************************************************
+
+To be able to track added and removed embedded entities across the update, all the
+relevant lifecycle states (i.e. all states traversed during an update) need to be
+provided with the following attributes:
+
+* previous_attr_set_on_validate
+* previous_attr_set_on_export
+
+The domain of valid values for these attributes is [``"candidate"``, ``"active"``, ``"rollback"``, ``null``].
+
+
+
+1. During a validation compile:
+    * Current set
+        - if the current instance is being validated:
+             - The current set is the set defined by the ``validate_self`` parameter.
+
+        - if another instance is being validated:
+             - The current set is the set defined by the ``validate_others`` parameter.
+
+    * Previous set
+        - The previous set is the set defined by the ``previous_attr_set_on_validate`` parameter.
+
+2. During an exporting compile:
+    * Current set
+        - The current set is the active attributes set
+
+    * Previous set
+        - The previous set is the set defined by the ``previous_attr_set_on_export`` parameter.
+
+
+We recommend to manually go through all the states that are part of the update, think about which
```

Comment:
I would add the actual method we used as well

In order to analyze a lifecycle, make a table as follows:

collumns: state, validation, previous attributes, current attributes, is_like, operation to is_like.

1. add one line for each state that is involved on an update or rollback scenario 
1. add the states just before and after the update (e.g. up and up)
1. add an additional line for each state that has a validating transfer 
    1. for the validating line, set `is_like` to the state we validating for (which state do we pretend to be, for which state do we want to know where are valid). Usually this state is soon after the validating transfer
    2. for the non-validating line for this transfer, set `is_like` to the state we try to maintain while validating. This is usually the state immediately before
    3. set the operations performed (promote, rollback) between the state and the `is_like` in `operation to is_like`  (for the non-validating line, this should be no operation)
1. for the main state in which the update is performed, set the `current attributes` to `active_attributes`  and the `previous_attributes ` to either `rollback attributes` for an update scenario and `candidate attributes` for a rollback scenario
2. for states with `is_like` set, apply the operations in reverse to the state they are like and previous attributes and  current attributes appropriately 


I would work this out for the simple lifecycle and add method and the example to the docs.

What I describe may be incomplete.